### PR TITLE
Migrate from posix to bash

### PR DESCRIPTION
**Pull request by brujoand**
**Migrate from busybox to bash**

This PR tries to fix som parsing mistakes, but adds a dependency on bash.

**Benefits**

The code was a bit hard to read and not posix compliant. I found that the scripts did not work for my cluster and decided to rewrite the utility scripts to bash and take care of possible causes of errors. Luckily that fixed my problem, so thus this PR.

**Possible drawbacks**

Now bash is a dependency, but in fairness it's only about 1mb so this shouldn't have much impact.

**Additional information**
I tried to avoid changing the flow of the scripts, but rather focus on getting them to work in my cluster.
With the modifications in this PR everything works for me locally. Later improvements could be to use proper arrays for the list of IPs and such. 
There is a lot of stuff going on in these scripts so give me a shout if I've missed something :)

### CODE REVIEW

File: bin/client_init.sh [hunk 1]
```diff
@@ -33,34 +35,33 @@ ip addr
...
+NAT_ENTRY="$(grep "$(hostname)" /config/nat.conf || true)"
 VXLAN_GATEWAY_IP="${VXLAN_IP_NETWORK}.1"
 
 # For debugging reasons print some info
 ip addr
 ip route
 
 # Check we can connect to the GATEWAY IP
-ping -c1 $GATEWAY_IP
+ping -c1 "$GATEWAY_IP" || true
```
Comment (angelnu) [inline bin/client_init.sh:48]: This test is intended to fail if we cannot reach the gateway so a retry can be done - could you please explain why we should ignore this?

File: bin/client_init.sh [hunk 2]
```diff
@@ -70,26 +71,25 @@ interface \"vxlan0\"
...
+  route add default gw "$VXLAN_GATEWAY_IP"
 fi
 
 # For debugging reasons print some info
 ip addr
 ip route
 
 # Check we can connect to the gateway ussing the vxlan device
-ping -c1 $VXLAN_GATEWAY_IP
+ping -c1 "$VXLAN_GATEWAY_IP" || true
```
Comment (angelnu) [inline bin/client_init.sh:93]: same comment here

File: bin/client_init.sh (hunk 1 above)
Comment (brujoand) [reply bin/client_init.sh:48]: Ah, remnant of my debugging. Will remove the catch-all.

**Review by angelnu — CHANGES_REQUESTED** [review]
This is a great rework - thanks!

I only have questions about the ignored ping errors to clarify before merging.

**Comment by brujoand** [comment]
Got the `|| true` out of the way now and re-checked that the deployment worked. I had some problems with ICMP packet specifically in the beginning so I bypassed them to be able to debug other parts of the scripts.

**Comment by angelnu** [comment]
Thanks @brujoand - could you please confirm it works for you with the dev tag? Then I would create a new official release.

**Comment by brujoand** [comment]
@angelnu Just tried with `ghcr.io/k8s-at-home/pod-gateway:dev@sha256:956f271004f72b2e5a0fd011ebedde6e31422ae7c8ae5c3203d0b7773f472624` and that worked fine 👍

**Comment by angelnu** [comment]
Testing on my end as well - will release the new version. Thanks @brujoand

**Comment by brujoand** [comment]
@angelnu ah, actually, I think there is a problem whith the heredoc in client init. The indentation is lost. Fixing now.

**Comment by brujoand** [comment]
@brujoand Did the dynamic ip work prior to this pr? It looks like some of the declarations in the generated dhclient.conf file is not supported like https://linux.die.net/man/5/dhclient.conf but they appear in the openbsd version https://man.openbsd.org/dhclient.conf.5 And the formatting doesn't look right as there are 2 lines commented out which actually removes the line ending semicolon. Looking at the generated dhclient.conf from after and before this PR they look identical, so I'm assuming this problem is unrelated.

**Comment by angelnu** [comment]
yes, the dhcp was working - it is actually the default.

I think I see the problem: the cat has to append to the /etc/dnsmasq.conf not replace it.

**Comment by brujoand** [comment]
> I think I see the problem: the cat has to append to the /etc/dnsmasq.conf not replace it.

Ah, right you are. Sorry about that, PR coming right up. I changed it to append rather than overwrite and ofc that fixed the problem. What caught my attention was these error messages:



```
/etc/dhclient.conf line 3: semicolon expected.
link-timeout 10;
              ^
/etc/dhclient.conf line 4: semicolon expected.
reboot
 ^
/etc/dhclient.conf line 15: no option named require in space dhcp
  require routers,
   ^
/etc/dhclient.conf line 18:  : expected option name.
 }
  ^
/etc/dhclient.conf line 18: unterminated interface declaration.
 }
```

Which are due to problems with the dhclient.conf, but I just checked and they were also there before this PR.

**Pull request merged**